insalan/payment/views.py

```diff
@@ -71,30 +71,62 @@ def post(self, request):
         if not data.get("metadata") or not data["metadata"].get("uuid"):
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
+        # Access to .metadata.uuid is verified
         uuid = data["metadata"]["uuid"]
         try:
             trans_obj = Transaction.objects.get(id=uuid)
         except Transaction.DoesNotExist:
             logger.error("Unable to find transaction %s", uuid)
             return Response(status=status.HTTP_400_BAD_REQUEST)
 
-        ntype = data["eventType"]
-        data = data["data"]
-
-        # logger.debug("NTYPE: %s", ntype)
-        # logger.debug("DATA: %s", data)
+        ntype = data.get("eventType")  # defaults to None
+        data = data.get("data", default={})
 
         if ntype == "Order":
             # From "Order", get the payments
-            order_id = data["id"]
+            # Check that the id is an integer
+            order_id = data.get("id")
+            if not isinstance(order_id, int):
+                logger.error(
+                    "Invalid ID for Order event, except integer, got %s", order_id
```

Comment:
Can be a bit ambiguous if it is a string containing only digits as it won't print quotes